compiler/rustc_error_codes/src/error_codes/E0404.md

```diff
@@ -30,14 +31,16 @@ struct Bar;
 impl Foo for Bar { // ok!
     // functions implementation
 }
+
+fn baz<T: Foo>(t: T) {} // ok!
 ```
 
-or:
+or, if you want to use a trait alias, not a type alias, you should use
+`#![feature(trait_alias)]`:
```

Comment:
I would change this to be something along the lines of

    Alternatively, you could introduce a new trait with your desired restrictions as a super trait:
    
    ```
    # trait Foo {}
    # struct Bar;
    # impl Foo for Bar {}
    trait Qux: Foo {} // Anything that implements Qux also needs to implement Foo
    fn baz<T: Qux>(t: T) {} // also ok!
    ```
    
    Finally, if you are on nightly and want to use a trait alias instead of a type alias, you should use `#![feature(trait_alias)]`:

Reply:
Thanks, just applied your suggestion.